src/Twig/SourceCodeExtension.php

```diff
@@ -41,10 +54,28 @@ public function setController(?callable $controller): void
     public function getFunctions(): array
     {
         return [
-            new TwigFunction('show_source_code', [$this, 'showSourceCode'], ['is_safe' => ['html'], 'needs_environment' => true]),
+            new TwigFunction('link_source_file', $this->linkSourceFile(...), ['is_safe' => ['html'], 'needs_environment' => true]),
+            new TwigFunction('show_source_code', $this->showSourceCode(...), ['is_safe' => ['html'], 'needs_environment' => true]),
         ];
     }
 
+    /**
+     * Render a link to a source file.
+     */
+    public function linkSourceFile(Environment $twig, string $file, int $line): string
+    {
+        $text = str_replace('\\', '/', $file);
+        if (str_starts_with($text, $this->projectDir)) {
+            $text = mb_substr($text, \mb_strlen($this->projectDir));
```

Comment:
No need to use mbstring here to remove the string from the start (we know it already starts with it): `substr($text, \strlen($this->projectDir))` works just fine (and would work even if the filesystem path is not in UTF-8)